skimage/morphology/_rolling_ball_cy.pyx

```diff
@@ -0,0 +1,34 @@
+import numpy as np
+cimport numpy as np
+
+cimport cython
+
+ctypedef np.float_t DTYPE
```

Comment:
Good question. I hadn't thought about NaNs.

I was thinking about them now, and, at first, I thought it could swallow/ignore NaNs as far as possible; however, after I thought about it more, I think it is more sensible to propagate them and set the affected area to NaN. That way, if NaN is important, it will propagate, and if it isn't then one can do something along the lines of `img[np.isnan[img]] = sensible_value`.

Reply:
Sounds good. Could you add a short note describing this behavior to the `Notes` section of each of the public function's docstrings?

Reply:
I tested this idea to propagate NaN. It makes the code about 4 times slower on my small test case. My new idea is to have two versions of `apply_kernel`; one that assumes there are no NaNs (fast), and one that assumes there are NaNs. I will also add a `has_nan` input argument.

Reply:
How about replacing any `NaN` values with `np.max(img)` before processing and replacing the same coordinates with `NaN` after processing? That way, `NaN` will be effectively ignored by the rolling ball without loosing much speed.

Reply:
Yes, that was my line of thinking in the beginning, too, @thawn . The problem is that we can't tell if `np.max(img)` is a good substitution. It is equivalent to saying that `np.min(tmp) < np.nan`, which can make sense in some cases but not in others. Instead of making any assumption, I think it will be easier to hold the user accountable for substituting NaN where necessary.

Reply:
@lagru Like this?
https://github.com/FirefoxMetzger/scikit-image/blob/4b955230a02993fe2d6a0a41f637b1a0774db989/skimage/morphology/rolling_ball.py#L25

Reply:
Yeah, that looks like a great solution! And I agree with your view that we can't make an assumption about the value of NaNs. See also https://github.com/scikit-image/scikit-image/pull/4851/files#r464668533.